### PR TITLE
Chapter 24 HTTP and Background Tasks

### PR DESCRIPTION
**Pull request by alexandragnj**
## New features

- New app called PhotoGallery which will display the most interesting public photo of the day from Flickr

## Changes in code

- Adding a fragment that holds a RecyclerView
- Adding a Retrofit API interface to define API calls that is fetching JSON from Flickr
- Creating an instance of the API using Retrofit object
- Getting a Call request and executing the request
- Deserialising JSON text into Kotlin using Gson
- Creating a model object class
- Making the objects into JSON data
- Adding a ViewModel class

## Screenshot 
Screenshot 2022-06-10 at 08 58 08

### CODE REVIEW

File: app/src/main/java/com/example/photogalery/PhotoAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class PhotoAdapter(private val galleryItem: List<GalleryItem>) :
+    RecyclerView.Adapter<PhotoAdapter.PhotoHolder>() {
+
+    class PhotoHolder(itemTextView: TextView) : RecyclerView.ViewHolder(itemTextView) {
+
+        val bindTitle: (CharSequence) -> Unit = itemTextView::setText
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhotoHolder {
+        val textView=TextView(parent.context)
```
Comment (ghost) [inline app/src/main/java/com/example/photogalery/PhotoAdapter.kt]: The file is not properly formatted

File: app/src/main/java/com/example/photogalery/PhotoGalleryActivity.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+
+class PhotoGalleryActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_photo_gallery)
+
+        val isFragmentContainerEmpty=savedInstanceState==null
```
Comment (ghost) [inline app/src/main/java/com/example/photogalery/PhotoGalleryActivity.kt]: The file is not properly formatted

**Review by ghost — COMMENTED** [review]
Looks good, just a couple formatting issues. Also don't forget to add `ktlint` and `detekt` to the app too

File: app/src/main/java/com/example/photogalery/PhotoAdapter.kt (hunk 1 above)
Comment (alexandragnj) [reply app/src/main/java/com/example/photogalery/PhotoAdapter.kt]: Done.

File: app/src/main/java/com/example/photogalery/PhotoGalleryActivity.kt (hunk 2 above)
Comment (alexandragnj) [reply app/src/main/java/com/example/photogalery/PhotoGalleryActivity.kt]: Done.

**Review by iugaliviu — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**